ocs_ci/ocs/cluster.py

```diff
@@ -1668,6 +1669,47 @@ def check_ceph_health_after_add_capacity(
     ), "Data re-balance failed to complete"
 
 
+def validate_osd_pdb():
+    """
+    Validate creation of PDBs for OSDs.
+    1. Versions lesser than ocs-operator.v4.6.2 have PDBs for each osds
+    2. Versions greater than or equal to ocs-operator.v4.6.2-233.ci have
+    PDBs collectively for osds like rook-ceph-osd
+
+    Returns:
+        bool: True if blocking PDBs are present, false otherwise
+
+    """
+    pdb_obj = ocp.OCP(
+        kind=constants.POD_DISRUPTION_BUDGET, namespace=defaults.ROOK_CLUSTER_NAMESPACE
+    )
+    pdb_obj_get = pdb_obj.get()
+    osd_pdb = []
+    for pdb in pdb_obj_get.get("items"):
+        if (
+            "rook-ceph-mds-ocs-storagecluster-cephfilesystem"
+            not in pdb["metadata"]["name"]
+            and "rook-ceph-mon-pdb" not in pdb["metadata"]["name"]
+        ):
+            osd_pdb.append(pdb["metadata"]["name"])
+    blocking_pdb_exist = False
+    if len(osd_pdb) == 1:
+        logger.info(f"No blocking PDBs created {osd_pdb}")
+    else:
+        logger.info(f"Blocking PDBs are created {osd_pdb}")
+        for osd in range(len(osd_pdb)):
```

Comment:
you dont need to iterate it over range, since it is a list you can directly iterate it as for osd in osd_pdb:"

Reply:
This leads to type error

>           for osd in len(osd_pdb):
E           TypeError: 'int' object is not iterable